### ISSUE TITLE
Question: Is it safe to start another thread logger in dlopened library?

### ISSUE THREAD

**Issue by evanzd** [odygrd/quill#301]
I want to use quill in both my main program as well as a dlopened library. However, as the dlopened library cannot access the static global logger instance in the main program, it has to start its own thread logger. So there will be two quill backend loggers running. Is it safe to use quill like this?

**Comment by odygrd** [comment]
I haven't tried to use it like this.
`quill::start()` will spawn a thread once and then if you call it again it won't do anything because it is using `std::call_once`

Can you try to call `quill::start()` in your main program as well and see if it spawns the thread ? As long as it spawns it then it should be fine

**Comment by evanzd** [comment]
I attempted to call quill::start() in both the main program and the dlopened library, which resulted in two `Quill_Backend` threads. Is this behavior normal?
<img width="363" alt="image" src="https://github.com/odygrd/quill/assets/10608479/29db8964-3fcd-483e-86d3-6341e11461a5">

**Comment by odygrd** [comment]
yes the logging thread spins see also https://github.com/odygrd/quill/discussions/275

You can pin the backend logging thread to a non critical core and also configure a sleep duration when it goes idle before calling `quill::start()`

This file has all the config options (https://github.com/odygrd/quill/blob/master/quill/include/quill/Config.h)

e.g.

```c++

  quill::Config cfg;
  cfg.backend_thread_sleep_duration = std::chrono::nanoseconds{500};
  cfg.backend_thread_cpu_affinity = 1; 

  quill::configure(cfg);
  quill::start();
```

It is also possible to set a high sleep duration and wake it up on demand from your hot threads but that will result in additional latency on your hot path see https://github.com/odygrd/quill/blob/master/quill/include/quill/Quill.h#L388
